File: android/src/main/kotlin/com/fzfstudio/ezw_ble/ble/models/BleUUID.kt
package com.fzfstudio.ezw_ble.ble.models

import com.fzfstudio.ezw_ble.ble.models.enums.BleUuidType
import java.io.Serializable
import java.util.UUID

data class BleUuid(
    val service: String,
    val writeChars: String? = null,
    val readChars: String? = null,
    val type: BleUuidType = BleUuidType.COMMON
): Serializable {

    val serviceUUID: UUID
        get() = UUID.fromString(service)
    val writeCharsUUID: UUID?
        get() = if (writeChars == null) null else UUID.fromString(writeChars)
    val readCharsUUID: UUID?
        get() = if (readChars == null) null else UUID.fromString(readChars)

}

File: ios/Classes/ble/models/BleUUID.swift
//
//  BleUUID.swift
//  EvenConnect
//
//  Created by Whiskee on 2025/1/6.
//

import CoreBluetooth

struct BleUuid: Codable {
    let service: String
    var writeChars: String?
    var readChars: String?
    var type: BleUuidType = BleUuidType.common
 
    // 计算属性，允许访问 CBUUID 对象
    var serviceUUID: CBUUID {
        get { return CBUUID(string: service) }
    }
    var writeCharUUID: CBUUID? {
        get { return writeChars.map(CBUUID.init(string:)) }
    }
    var readCharUUID: CBUUID? {
        get { return readChars.map(CBUUID.init(string:)) }
    }
    
}
